backend/error_source.go

```diff
@@ -0,0 +1,41 @@
+package backend
+
+import "net/http"
+
+// ErrorSource type defines the source of the error
+type ErrorSource string
+
+const (
+	ErrorSourcePlugin     ErrorSource = "plugin"
+	ErrorSourceDownstream ErrorSource = "downstream"
+)
+
+type PluginError struct {
```

Comment:
I would hold of with introducing this struct until we've investigated/figured out returned errors part of https://github.com/grafana/grafana/issues/75761. Alternatively I would suggest adding it to experimental if you really need it.